pkg/platform/api/graphql/model/buildplanner/buildExpression.go

```diff
@@ -151,51 +190,101 @@ func (bx *BuildExpression) Update(operation Operation, requirement Requirement)
 	return nil
 }
 
-func (bx *BuildExpression) AddRequirement(requirement Requirement) error {
-	bx.requirementsNode = append(bx.requirementsNode, requirement)
+// addRequirement adds a requirement to the BuildExpression.
+func (bx *BuildExpression) addRequirement(requirement Requirement) error {
+	bx.requirements = append(bx.requirements, requirement)
 
-	(*bx.solveNode)[RequirementsKey] = bx.requirementsNode
+	(*bx.solveNode)[RequirementsKey] = bx.requirements
```

Comment:
It _shouldn't_ be but I don't believe there is any rule that states that. Added a nil check just to be sure.

Reply:
It can't be due to the check for a valid requirements node: https://github.com/ActiveState/cli/pull/2559/files/c59dcb70e0864748059fce204b9d7e5fe12b825f..3c966441e848668f584cfd5871c3f4208597cdf2#diff-733c58be1cfc77e97f562b5cf09693ccecbf956fd0f3accf8cd9c3ab40d59b09R287-R301

In a previous review I had requested that a BuildExpression validate its incoming JSON to error as early as possible.